mllib/src/main/scala/org/apache/spark/mllib/clustering/KMeans.scala

```diff
@@ -292,15 +292,16 @@ class KMeans private (
       }.reduceByKey { case ((sum1, count1), (sum2, count2)) =>
         axpy(1.0, sum2, sum1)
         (sum1, count1 + count2)
+      }.map { case (j, (sum, count)) =>
+        scal(1.0 / count, sum)
```

Comment:
Sorry, I've just thought of a further simplification:

```
.mapValues { case (sum, count) =>
  scal(1.0 / count, sum)
  new VectorWithNorm(sum)
}.collectAsMap()
```

I'll run the tests though